xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/Timers.h

```diff
@@ -0,0 +1,816 @@
+         i < m_cStructure->iLifetimesSize && i < sizeof(m_cStructure->lifetimes); ++i)
+    {
+      m_cStructure->lifetimes[i].iValue = lifetimes[i].first;
+      strncpy(m_cStructure->lifetimes[i].strDescription, lifetimes[i].second.c_str(),
+              sizeof(m_cStructure->lifetimes[i].strDescription) - 1);
+    }
+    if (lifetimesDefault != -1)
+      m_cStructure->iLifetimesDefault = lifetimesDefault;
+  }
+
+  /// @brief To get with @ref SetLifetimes changed values.
+  std::vector<std::pair<int, std::string>> GetLifetimes() const
+  {
+    std::vector<std::pair<int, std::string>> ret;
+    for (unsigned int i = 0; i < m_cStructure->iLifetimesSize; ++i)
+    {
+      int value = m_cStructure->lifetimes[i].iValue;
+      std::string description = m_cStructure->lifetimes[i].strDescription;
+      ret.push_back(std::pair<int, std::string>(value, description));
+    }
+    return ret;
+  }
+
+  /// @brief **optional**\n
+  /// The default value for @ref SetLifetimes().
+  ///
+  /// @note Must be filled if @ref SetLifetimes contain values and not
+  /// defined there on second function value.
+  void SetLifetimesDefault(int lifetimesDefault)
+  {
+    m_cStructure->iLifetimesDefault = lifetimesDefault;
+  }
+
+  /// @brief To get with @ref SetLifetimesDefault changed values.
+  int GetLifetimesDefault() const { return m_cStructure->iLifetimesDefault; }
+
+  /// @brief **optional**\n
+  /// Prevent duplicate episodes value definitions.
+  ///
+  /// Array containing the possible values for @ref PVRTimer::SetPreventDuplicateEpisodes().
+  ///
+  /// @note Must be filled if @ref PVRTimer::SetPreventDuplicateEpisodes() is not empty.
+  ///
+  /// @param[in] preventDuplicateEpisodes List of duplicate episodes values
+  /// @param[in] preventDuplicateEpisodesDefault [opt] The default value in list, can also be
+  ///                                            set by @ref SetPreventDuplicateEpisodesDefault()
+  void SetPreventDuplicateEpisodes(
+      const std::vector<std::pair<int, std::string>>& preventDuplicateEpisodes,
+      int preventDuplicateEpisodesDefault = -1)
+  {
+    m_cStructure->iPreventDuplicateEpisodesSize = preventDuplicateEpisodes.size();
+    for (unsigned int i = 0; i < m_cStructure->iPreventDuplicateEpisodesSize &&
+                             i < sizeof(m_cStructure->preventDuplicateEpisodes);
+         ++i)
+    {
+      m_cStructure->preventDuplicateEpisodes[i].iValue = preventDuplicateEpisodes[i].first;
+      strncpy(m_cStructure->preventDuplicateEpisodes[i].strDescription,
+              preventDuplicateEpisodes[i].second.c_str(),
+              sizeof(m_cStructure->preventDuplicateEpisodes[i].strDescription) - 1);
+    }
+    if (preventDuplicateEpisodesDefault != -1)
+      m_cStructure->iPreventDuplicateEpisodesDefault = preventDuplicateEpisodesDefault;
+  }
+
+  /// @brief To get with @ref SetPreventDuplicateEpisodes changed values.
+  std::vector<std::pair<int, std::string>> GetPreventDuplicateEpisodes() const
```

Comment:
The std::pair here acts as a simple C++ wrapper for the C API struct PVR_ATTRIBUTE_INT_VALUE. May I suggest to introduce a dedicated C++ wrapper class for future safeness? If we are going to extend PVR_ATTRIBUTE_INT_VALUE by a third attribute, we must change this method's return value as a pair does not work anymore, then. And every calling code would need to be adapted, even if it is not using the new attribute. Applies to all new C++ wrapper methods returning std::pair.